router/src/namespace_cache/memory.rs

```diff
@@ -56,12 +56,13 @@ impl NamespaceCache for Arc<MemoryNamespaceCache> {
             Some(old) => merge_schema_additive(schema, old),
             None => {
                 let change_stats = ChangeStats {
-                    new_columns: schema
-                        .tables
-                        .values()
-                        .map(|v| v.column_count())
-                        .sum::<usize>(),
-                    new_tables: schema.tables.len(),
+                    new_tables: schema.tables.clone(),
+                    // There are no pre-existing tables for columns to be added
+                    // to, so don't need to build another map.
+                    new_columns_per_table: Default::default(),
+                    num_new_columns: schema.tables.iter().fold(0, |acc, (_, table_schema)| {
+                        acc + table_schema.column_count()
+                    }),
```

Comment:
I think this will probably be faster:

```suggestion
                    num_new_columns: schema.tables.iter().map(|(_, v)| v.column_count()).sum(),
```

I dunno if the former will vectorise, but this definitely will.